README.md

```diff
@@ -41,3 +41,46 @@ spec:
 # Screenshots
 
 ![picture](screenshot.png)
+
+# JSONL reader
+
+## API
+
+This functionality needs `th2-json-stream-provider 0.0.7 (or newer)` backend component to function. It should be accessible at `{rpt-viewer-path}/json-stream-provider*`
+
+## Features
+
+- It able to view contents of JSONL files, such files could be loaded from local or from server using  `th2-json-stream-provider`. It will display complex values as tree leafs and simple values of selected leaf would be visible in table. Amount indicator of complex fields will be `{N}` after leaf name and for simple fields it would be in `(N)`. If leaf has field `#display-timestamp` with numeric timestamp value in nanoseconds it would be converted to human readable format and displayed after indicators.
+- View type could be changed using 3 dots on the right of the leaf. It has 4 types:
+  - Plain JSON in one line;
+  - Formated JSON;
+  - Tree view;
+  - Table, which will display field `#display-table`, which is 2D Array where first array is Header and others is rows, as formated table below leaf;
+- View could be changed for whole group via buttons on the right of group name
+- Leaf with `#display-timestamp` will be colored depending on current interval which could be changed at the top of view. It has millisecond, second, minute types of interval and any positive numeric value could be entered for them.
+- It has search which will highlight searched values in the viewing files. Current search could be exported in the JSON file, and imported from such files. Search files has this structure
+
+```
+[{
+  "pattern": "searched value",
+  "color": "hex color"
+}, ...]
+```
+
+- JSONL reader can also get data of Jupyter Notebooks using  `th2-json-stream-provider`, when notebook is requested it will be displayed as list of parameters which could be modified. Then this notebook could be launched and resulting JSONL file will be requested on finish and added to current list.
+- Notebooks has default result of 1 in which case new results will overwrite previous result of this notebook, this could be changed in the bottom of notebook editing form, if amount is more than 1 then result will overwrite oldest one.
```

Comment:
```suggestion
- Notebooks tab holds one one result by default in which case new results will overwrite previous result of this notebook, this could be changed in the bottom of notebook editing form, if amount is more than 1 then result will overwrite oldest one.
```